### PR TITLE
Gearpump 257 remote application submit 

### PR DESCRIPTION
**Pull request by th3iedkid**
https://issues.apache.org/jira/browse/GEARPUMP-257 
 `[GEARPUMP-Gearpump 257] Client context should throw an exception when client tries to submit an application with jar file that does not exist. 
 Previously it used to upload an empty stream and master used to fail with file not found exception.

### CODE REVIEW

**Comment by manuzhang** [comment]
@th3iedkid  Could you reformat your commit message to `[GEARPUMP-257] Throw an exception when client tries to submit an application with jar file that does not exist`. 

R: @huafengw

**Comment by th3iedkid** [comment]
@manuzhang , have edited the commit message and now i can it  appear in JIRA too!

File: core/src/main/scala/org/apache/gearpump/jarstore/FileServer.scala [hunk 1]
```diff
@@ -145,8 +145,7 @@ object FileServer {
     }
 
     private def entity(file: File)(implicit ec: ExecutionContext): Future[RequestEntity] = {
-      val entity = HttpEntity(MediaTypes.`application/octet-stream`, file.length(),
-        FileIO.fromFile(file, chunkSize = 100000))
+      val entity = HttpEntity(MediaTypes.`application/octet-stream`, file, chunkSize = 100000)
```
Comment (manuzhang) [inline core/src/main/scala/org/apache/gearpump/jarstore/FileServer.scala]: Why do you need to change this file ? I don't see it necessary here.

Comment (th3iedkid) [inline core/src/main/scala/org/apache/gearpump/jarstore/FileServer.scala]: yes its unconnected ,shall remove it if you feel so ,
However was wondering if it made the method more focused & better performant on setting up entity with file directly using Akka http API than us us doing a FileIO ?
The akka http API does the same plus 
1-  we don't fetch the length of the file ourselves ,
2- short-circuits if the file length were zero , prevent graph orchestration possibly for File IO reads of empty files.

Believe these can be useful for a client context.

Let me know otherwise , shall undo.

Comment (manuzhang) [inline core/src/main/scala/org/apache/gearpump/jarstore/FileServer.scala]: That looks valid but could you put it in a separate PR ?

Comment (th3iedkid) [inline core/src/main/scala/org/apache/gearpump/jarstore/FileServer.scala]: Have removed those changes now.
shall pull in another request for that .

**Comment by manuzhang** [comment]
LGTM, merging.
